articles/hapi-vs-koa-vs-express/index.md

```diff
@@ -0,0 +1,159 @@
+const hapi = require('hapi')
+const server = new hapi.Server()
+
+server.connection({
+    host: 'localhost',
+    port: '2400'
+})
+
+server.start(error => {
+    if (error) throw error
+    else console.log('Server running at PORT 2400');
+})
+```
+
+### Framework advantages and disadvantages
+In this section we are going to highlight some of the advantages and disadvantages of each framework.
+
+#### Hapi.js
+##### Advantages
+1. Hapi enables one to build a scalable web application or API.
+2. Hapi uses plugins which helps in scaling.
+3. Hapi can also be used in conjunction with front-end frameworks like Vue and React to build single-page web applications.
+4. Hapi provides a deeper control over request handling through the use of context.
+5. Hapi provides input/output validation, routing and caching, hence making it a good choice for building REST APIs.
+6. Hapi provides good plugins such as Yar, Good, Catbox etc.
+
+##### Disadvantages
+1. Refactoring in Hapi is manual.
+2. Endpoints must be created and tested manually.
+3. It restricts developers to using Hapi specific plugins and modules which are not compatible with other frameworks.
+
+#### Koa.js
+##### Advantages
+1. Koa has a small footprint hence making it very light weight.
+2. Koa has a good overall user experience.
+3. Koa uses try/catch which improves error handling.
+4. Koa has a cleaner and more readable asynchronous code.
+5. Koa does not use callbacks hence facilitating upstream and downstream flow of controls.
+
+##### Disadvantages
+1. Koa has a relatively small community.
+2. It is not compatible with middleware builds using express style.
+3. Koa uses ES6 generators which are not compatible with middleware from other framework.
+
+#### Express.js
+##### Advantages
+1. Express has a large community.
+2. Express is simple, flexible, and scalable.
+3. Express is fully customizable.
+4. It is easy to integrate third party services and middleware.
+5. Express has a low learning curve.
+
+##### Disadvantages
+1. Express requires a lot of manual labor.
+2. Refactoring bigger code bases is very challenging.
+
+### Framework performance
+In this section, we will compare the frameworks based on the number of request that a hello world server can handle in one second.
+
+Hapi.js handles 29,998 requests per second while Express handles 38,510 requests per second.
+
+Koa.js seems to be best performing since it handles 50,933 requests per second.
+
+### Choosing the right framework for your project
+
+#### Why choose Express.js
+Express provides a wide range of functionalities for developing web applications such as single/multi-page applications and even hybrid applications. Express is also easy to test due to its middleware and routes which modularize projects. Express.js framework is also closer to Node.js when compared to the other Frameworks, it is only a thin layer to Node.js hence making it the best for building applications that require significantly better performance.
```

Comment:
Remove framework in Express.js framework for conciseness. Lowercase frameworks. because instead of , afterwards. That require **the best performance** would flow better.